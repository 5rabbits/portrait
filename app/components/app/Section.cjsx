React     = require('react')

Loader    = require('../shared/Loader')
Offscreen = require('../shared/Offscreen')
EmptyView = require('../shared/EmptyView')

@Section = React.createClass
  render: ->
    return (
      <section className="section-wrapper" id="block-section">
        <header className="section-controls" id="block-controls">
          <div className="container-max">
            { @props.controls }
          </div>
        </header>
        <section className="section-content" id="block-content">
          <div className="container-max">

            <article className="hidden section-state" id="block-content-loading">
              <Loader/>
            </article>

            <article className="hidden section-state" id="block-content-empty">
              <EmptyView/>
            </article>

            <article className="hidden- section-state" id="block-content-data">
              { @props.content }
            </article>

          </div>
        </section>
      </section>
    )

@SectionOffscreen = React.createClass
  render: ->
    return (
      <section className="section-wrapper" id="block-section">
        <header className="section-controls" id="block-controls">
          <div className="container-max">
            { @props.controls }
          </div>
        </header>
        <section className="section-content" id="block-content">
          <div className="container-max">

            <Offscreen
              panel={
                <div className="container-page">
                  <form>
                    <div className="form-group">
                      <label className="type-caption" for="exampleInputEmail1">Email address</label>
                      <input type="email" className="form-control" id="exampleInputEmail1" placeholder="Email"/>
                    </div>
                    <div className="form-group">
                      <label className="type-caption" for="exampleInputPassword1">Password</label>
                      <input type="password" className="form-control" id="exampleInputPassword1" placeholder="Password"/>
                    </div>
                    <div className="checkbox">
                      <label>
                        <input type="checkbox"/> Check me out
                      </label>
                    </div>
                    <button type="submit" className="btn btn-default">Submit</button>
                  </form>
                </div>
              }
              content={
                <div className="section-states">

                  <article className="hidden section-state" id="block-content-loading">
                    <Loader/>
                  </article>
                  <article className="hidden section-state" id="block-content-empty">
                    <EmptyView/>
                  </article>
                  <article className="hidden- section-state" id="block-content-data">
                    { @props.content }
                  </article>

                </div>
            }/>

          </div>
        </section>
      </section>
    )

module.exports = @SectionOffscreen
