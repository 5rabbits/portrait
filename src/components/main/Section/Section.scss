@import '~styles/core';

.Section {
  min-height: 100%;
  display: flex;
  flex-direction: column;
}

.Section__toolbar {
  border-bottom: #DBE1E6 solid 1px;
  position: fixed;
  right: 0;
  left: 0;
  z-index: 2;
  flex-shrink: 0;
}

.Section__contents {
  padding-top: $spacing-base * 4;
  flex: 1;
  display: flex;
  align-items: stretch;
  flex-direction: column;
}

.Section--with-toolbar {
  .Section__contents {
    padding-top: $section-toolbar-height;
  }
}
