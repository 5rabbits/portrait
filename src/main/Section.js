import React from 'react'

const Section = props =>
  <section className="section-wrapper" id="block-section">
    <header className="section-controls" id="block-controls">
      <div className="container-max">
        {props.controls}
      </div>
    </header>
    <section className="section-content" id="block-content">
      <div className="container-max">
        {props.content}
      </div>
    </section>
  </section>

export default Section

