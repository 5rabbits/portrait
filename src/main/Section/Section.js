import React from 'react'
import Controls from 'main/Controls'
import childrenPropType from 'propTypes/children'

const Section = props =>
  <section className="section-wrapper" id="block-section">
    <header className="section-controls" id="block-controls">
      <div className="container-max">
        {props.controls}
      </div>
    </header>
    <section className="section-content" id="block-content">
      <div className="section-subhead">
        <div className="container-max">
          <Controls
            main={<a href="#" className="link link-badge type-footnote"><i className="fa fa-filter" /><span className="ml-xs hidden-xs">Ver filtros</span></a>}
            content={<span className="type-subhead">Hay <strong>356</strong> resultados</span>}
            actions={<a href="#" className="link link-badge type-footnote"><span className="mr-xs hidden-xs">Descargar listado</span><i className="fa fa-download" /></a>}
          />
        </div>
      </div>

      <div className="container-max">
        {props.content}
      </div>
    </section>
  </section>

Section.propTypes = {
  content: childrenPropType,
  controls: childrenPropType,
}

export default Section
