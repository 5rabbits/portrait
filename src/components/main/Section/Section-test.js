import { React, expect, mount } from 'test/helper'
import Section from 'main/Section'

describe('Section', () => {
  it('should render with minimum props', () => {
    const component = mount(<Section />)

    expect(component).to.be.ok
  })
})
