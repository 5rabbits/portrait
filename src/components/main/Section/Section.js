import React, { PureComponent, PropTypes } from 'react'
import cx from 'classnames'
import './Section.scss'

export default class Section extends PureComponent {
  static propTypes = {
    children: PropTypes.node.isRequired,
    toolbar: PropTypes.node,
  }

  static defaultProps = {
    toolbar: null,
  }

  render() {
    const { children, toolbar } = this.props

    return (
      <section
        className={cx('Section', {
          'Section--with-toolbar': toolbar,
        })}
        >
        {toolbar &&
          <header className="Section__toolbar">
            {toolbar}
          </header>
        }
        <section className="Section__contents">
          {children}
        </section>
      </section>
    )
  }
}
