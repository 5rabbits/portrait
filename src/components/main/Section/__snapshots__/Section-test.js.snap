// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section should allow to render a toolbar if specified 1`] = `
<Section
  className={null}
  toolbar="TOOLBAR"
>
  <section
    className="Section Section--with-toolbar"
  >
    <header
      className="Section__toolbar"
    >
      TOOLBAR
    </header>
    <section
      className="Section__contents"
    >
      Test
    </section>
  </section>
</Section>
`;

exports[`Section should render with minimum props 1`] = `
<Section
  className={null}
  toolbar={null}
>
  <section
    className="Section"
  >
    <section
      className="Section__contents"
    >
      Test
    </section>
  </section>
</Section>
`;
